### PR TITLE
add requestTimeout to tweek-client

### PR DESCRIPTION
**Pull request by oreporan**
added new config param `requestTimeoutInMillis` with default to 8sec. 
Runs `Promise.race()` with the native fetch

### CODE REVIEW

File: js/tweek-client/src/utils.ts [hunk 1]
```diff
@@ -2,6 +2,16 @@ export function captialize(string) {
   return string.charAt(0).toUpperCase() + string.slice(1);
 }
 
+export const requestTimeout = (timeoutInMillis): Promise<Response> => {
+  const failureResponse = new Response(null, { status: 408 });
+  return new Promise((res, rej) => {
+    let wait = setTimeout(() => {
+      clearTimeout(wait);
```
Comment (nataly87s) [inline js/tweek-client/src/utils.ts]: no need for `clearTimeout` at this point

File: js/tweek-client/src/utils.ts [hunk 2]
```diff
@@ -2,6 +2,16 @@ export function captialize(string) {
   return string.charAt(0).toUpperCase() + string.slice(1);
 }
 
+export const requestTimeout = (timeoutInMillis): Promise<Response> => {
```
Comment (nataly87s) [inline js/tweek-client/src/utils.ts]: maybe make this `(timeout, fetch) => (input, init) => Promise.race(...)` to create a fetch with timeout

File: js/tweek-client/src/utils.ts [hunk 3]
```diff
@@ -2,6 +2,19 @@ export function captialize(string) {
   return string.charAt(0).toUpperCase() + string.slice(1);
 }
 
+export const fetchWithTimeout = (timeout, fetch): Promise<Response> => {
```
Comment (nataly87s) [inline js/tweek-client/src/utils.ts]: I thought something more like this:
```
const createFetchWithTimeout = (timeout, fetch) => (input, init) => {
  return Promise.race([fetch(input, init), requestTimeout(timeout)]);
}
```
and then use it like this:
```
fetchClient = createFetchWithTimeout(requestTimeoutInMillis, fetchClient )
```

Comment (oreporan) [reply js/tweek-client/src/utils.ts]: Well, I think this way "fetch" is more abstract, I just wrap it in a Race with timeout...I don't care what params you give it..

Comment (oreporan) [reply js/tweek-client/src/utils.ts]: Anyways I need this change to be merged for duty issues...So we can do the 2nd interation later

Comment (nataly87s) [reply js/tweek-client/src/utils.ts]: Fetch is a standard spec, multiple environments are implementing it like node, browser vendors and RN native. That's the reason we are using it. So it's ok to expect there parameters

Comment (nataly87s) [reply js/tweek-client/src/utils.ts]: I don't think 5 more minutes will make that big of a difference. We can make these changes before merging

**Pull request merged**